### PR TITLE
more sync fixes

### PR DESCRIPTION
**Pull request by czue**
probably easiest read commit by commit. i ended up rewriting (and hopefully significantly simplifying) the new sync code.

wait for tests.

@snopoke cc @dannyroberts @benrudolph

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -651,55 +644,92 @@ def update_phone_lists(self, xform, case_list):
...
+                self.case_id = case_id
+                self.is_live = True
+                self.indices_to_add = []
+                self.indices_to_delete = []
+
+        ShortIndex = namedtuple('ShortIndex', ['case_id', 'identifier', 'referenced_id'])
+
+        # this is a variable used via closures in the function below
+        owner_id_map = {}
+        def get_latest_owner_id(case_id, action=None):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E301 expected 1 blank line, found 0

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -651,55 +644,92 @@ def update_phone_lists(self, xform, case_list):
...
-                    if log_has_case:
-                        # this case is being closed. we can try pruning this case since it's no longer relevant
-                        to_prune.add(case._id)
-                        made_changes = True
+                    case_update.is_live = False
 
-        for case_to_prune in to_prune:
-            self.prune_case(case_to_prune)
+        def _add_index(index):
+            logger.debug('adding index {} -> {} ({}).'.format(index.case_id, index.referenced_id, index.identifier))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E501 line too long (116 > 115 characters)

File: corehq/ex-submodules/casexml/apps/phone/models.py (hunk 1 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: (this was intentional since the variable is essentially part of the function)

**Comment by snopoke** [comment]
Nothing blocking from me, if any of my comments have value you can do a follow up PR.

**Pull request merged**